apps/apps.c

```diff
@@ -2340,6 +2340,72 @@ int app_isdir(const char *name)
 }
 #endif
 
+/* app_dirname section */
+
+/*
+ * This exactly follows what POSIX's
+ * dirname does, but is implemented
+ * in a more platform independent way.
+ *
+ * path        dirname
+ * /usr/lib    /usr
+ * /usr/       /
+ * usr         .
+ * /           /
+ * .           .
+ * ..          .
+ * ""          .
+ *
+ * Note: this function also keeps the
+ * possibility of modifying the 'path'
+ * string same as POSIX dirname.
+ */
+static char *posix_dirname(char *path)
+{
+    size_t l;
+    char *ret = ".";
+
+    l = strlen(path);
+    if (l == 0)
+        goto out;
+    if (strcmp(path, ".") == 0)
+        goto out;
+    if (strcmp(path, "..") == 0)
+        goto out;
+    if (strcmp(path, "/") == 0) {
+        ret = "/";
+        goto out;
+    }
+    if (path[l - 1] == '/')
+        /* /usr/ */
+        path[l - 1] = '\0';
```

Comment:
The comment effectively makes this multi-line. Please use {}

Reply:
Fixed